src/hapi/pipelines/database/locations.py

```diff
@@ -24,15 +25,37 @@ def __init__(
         )
         self.hapi_countries = configuration["HAPI_countries"]
         self.data = {}
+        self._datasetinfo = configuration["locations_hrp_gho"]
 
     def populate(self):
+        has_hrp, in_gho = self.read_hrp_gho_data()
         for country in Country.countriesdata()["countries"].values():
             code = country["#country+code+v_iso3"]
             location_row = DBLocation(
                 code=code,
                 name=country["#country+name+preferred"],
+                has_hrp=has_hrp.get(code, False),
+                in_gho=in_gho.get(code, False),
                 reference_period_start=parse_date(country["#date+start"]),
             )
             self._session.add(location_row)
             self._session.commit()
             self.data[code] = location_row.id
+
+    def read_hrp_gho_data(self):
+        has_hrp = {}
+        in_gho = {}
+        reader = Read.get_reader()
+        headers, iterator = reader.get_tabular_rows(
+            self._datasetinfo["url"],
+            headers=2,
+            dict_form=True,
+            format="csv",
+            file_prefix="locations",
+        )
+        for row in iterator:
+            if row["#indicator+hrp+bool"] == "Y":
```

Comment:
For code like this I would often force the case i.e. row["#indicator+hrp+bool"].upper() == "Y" because not doing so relies on humans to do a thing which they may not consider important.

Reply:
Good idea!